python/events/evaluation_tools/events/event_detection/decomposition.py

```diff
@@ -139,13 +178,36 @@ def mark_event_flows(
     # Eliminate negative values
     detrended[detrended < 0.0] = 0.0
 
-    # Return mask of non-zero detrended flows
-    return pd.Series((detrended > 0.0), index=series.index)
+    # Do not filter events
+    if minimum_event_duration == '0H':
```

Comment:
You could remove the code redundancy by moving the conditional after `event_points` is constructed and just return `event_points` if the criteria is met. 

As for `minimum_event_duration == '0H':`, I think we may want to make this comparison more robust. So for example, the 0 hour `datetime.timedelta` is caught. Why im here, im probably just misunderstanding, but should one of the accepted types of `minimum_event_duration` be `pd.tseries.offsets.DateOffset`? I had not come across that object in `pandas` before myself and after a [little digging](https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#overview) it looks like it is used to do calendar respective calculations, not temporally absolute (`pd.Timedelta`). Having played with a `pandas.tseries.offsets.DateOffset(hour=0)` , I can't find a way to easily get back a period string from the offset either just as a heads up. 

Also, a tuple as an excepted type? I am just a little confused. I checked that all of the object types you listed in the type hints can evaluate vs `==` without error, so that checks out. Just more curious than anything myself.

Reply:
Also having looked at how `event_points` is used later, namely `event_points.loc[:] = False`, it makes sense from the code readers perspective to change these names. Or just create an empty `pd.Series` with length `event_points`.

Reply:
OK. Thanks for this feedback. I changed `minimum_event_duration` to be a `pd.Timedelta` instead of an `offset`. This makes more sense. I also renamed the filtered series and generated a new series instead of re-assigning the values.